### PR TITLE
Call email user

### PR DESCRIPTION
**Pull request by MandyMeindersma**
Solves [#135](https://github.com/CMPUT301F16T01/Carrier/issues/135)

### CODE REVIEW

**Comment by MandyMeindersma** [comment]
I HAVE MADE A COMMIT THAT FIXES THE MERGE CONFLICTS BUT I WILL PUSH AFTER TO NOT OVERWEHLM YOU

**Comment by Hreherch** [comment]
Can you push the commit with the fixed conflicts? I need to see them to approve your branch :P

**Comment by MandyMeindersma** [comment]
READY TO REVIEW. MERGE CONFLICTS IN THERE TOO

**Comment by MandyMeindersma** [comment]
I MEAN NO MERGE CONFLICTS

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,16 +2,21 @@
...
     <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
     <uses-permission android:name="android.permission.INTERNET" />
+
+    <!-- USED FOR LOCATION -->
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_WIFI_STATE" />
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
 
+    <!-- USED TO CALL-->
```
Comment (sehoven) [inline app/src/main/AndroidManifest.xml]: Maybe say "USED TO MAKE PHONE CALLS". For a second I was thinking function calls :P

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java [hunk 2]
```diff
@@ -221,7 +219,7 @@ public void onRequestPermissionsResult(int requestCode,
                     AlertDialog.Builder adb = new AlertDialog.Builder(this);
                     adb.setTitle("Permissions Denied");
                     adb.setMessage("You cannot view the map to select locations without " +
-                            "allowing the app to access your device's storage. You can change " +
+                            "allowing the app to access. You can change " +
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java]: Why did you remove "your device's storage"? This is a grammary thing but I feel like the sentence doesn't make sense now...

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java [hunk 3]
```diff
@@ -255,4 +281,85 @@ public void hideKeyboard(View v) {
...
+
+    /**
+     * This is the function that calls the number presented
+     *
+     * @param v view used for this function
+     */
+    public void callPhone(View v) {
+        /* Source: http://stackoverflow.com/questions/5403308/make-a-phone-call-click-on-a-button
+        * Author: Shaista Naaz
+        * Retrieved on: November 21st 2016 */
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java:295]: Remind me to fix code attribution format. (If you want to see how I've been doing it, feel free to make the change yourself by looking at my code-reuse-fixes branch.) I can make the fix if you want once this is merged with master. Just maybe make an issue to remind me to deal with them or a TODO here or something :)

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java [hunk 4]
```diff
@@ -255,4 +281,85 @@ public void hideKeyboard(View v) {
...
+    /**
+     * This is the function that will email the email pressed
+     * @param v the view the function is in
+     */
+    public void emailUser(View v) {
+        /**
+         * Source: http://stackoverflow.com/questions/21720640/sending-email-from-android-app-when-click-on-button
+         * Author: localhost
+         * Retrieved on: November 21st 2016
+         */
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java:322]: Same thing as above with the code attribution format.

File: app/src/main/res/layout/activity_user_profile.xml [hunk 5]
```diff
@@ -99,6 +99,7 @@
                 android:textSize="24sp"
                 android:textColor="@android:color/white"
                 android:text="1234567"
+                android:onClick="callPhone"
                 android:focusable="false"
                 android:clickable="false"
                 android:background="@null"
```
Comment (sehoven) [inline app/src/main/res/layout/activity_user_profile.xml:110]: This is maybe a more stylistic thing but I think that as a user I was expecting to click on the phone and email images to make the call/email instead of the text. At the very least, maybe we could make both of them do the action? Idk...

Comment (sehoven) [reply app/src/main/res/layout/activity_user_profile.xml:110]: Also I have no idea how hard this would be and it is maybe an enhancement (could just make an enhancement issue for it) but the call/email action doesn't take place right away for me so some kind of visual clue that I pressed something clickable would be nice. Like you know when a button flashes or something? If someone knows how to do this now that would be a cool addition, but I don't think it's necessary per se.

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java [hunk 6]
```diff
@@ -255,4 +281,85 @@ public void hideKeyboard(View v) {
         InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
         imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
     }
+
+
+    /**
+     * This is the function that calls the number presented
+     *
+     * @param v view used for this function
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java]: Maybe for this javadoc just specify what view it is (i.e. what are you clicking on to call this function?)

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java [hunk 7]
```diff
@@ -255,4 +281,85 @@ public void hideKeyboard(View v) {
...
+            } else {
+                Toast.makeText(this, "You do not have the correct permissions to make a phone call.", Toast.LENGTH_SHORT).show();
+            }
+        }
+    }
+
+
+    /**
+     * This is the function that will email the email pressed
+     * @param v the view the function is in
```
Comment (sehoven) [inline app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java]: Same comment about the view as above.

**Review by sehoven — CHANGES_REQUESTED** [review]
Overall, it looks really good! A few tiny fixes, but nothing too drastic! Good work!

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (MandyMeindersma) [reply app/src/main/AndroidManifest.xml]: done

File: app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java (hunk 2 above)
Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/MainActivity.java]: fixed it

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java (hunk 6 above)
Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java]: changed

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java (hunk 3 above)
Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java:295]: made TODO

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java (hunk 4 above)
Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java:322]: made TODO

File: app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java (hunk 7 above)
Comment (MandyMeindersma) [reply app/src/main/java/comcmput301f16t01/github/carrier/Users/UserProfileActivity.java]: changed

File: app/src/main/res/layout/activity_user_profile.xml (hunk 5 above)
Comment (MandyMeindersma) [reply app/src/main/res/layout/activity_user_profile.xml:110]: added issue and changed button things to both being pressed

**Comment by MandyMeindersma** [comment]
I AM DONE

**Comment by sehoven** [comment]
I'm happy with everything here now. Please fix the merge conflicts and then I will merge this.

**Review by sehoven — APPROVED** [review]
